### ISSUE TITLE
wrong number of arguments calling `to_s` (1 for 0)

### ISSUE THREAD

**Issue by sunpeak** [logstash-plugins/logstash-codec-netflow#19]
"exception"=>#<ArgumentError: wrong number of arguments calling `to_s` (1 for 0)>, 
"backtrace"=>["/home/logstash/logstash/lib/logstash/codecs/netflow/util.rb:50:in `get'",
 "/home/logstash/logstash/vendor/bundle/jruby/1.9/gems/bindata-2.0.0/lib/bindata/array.rb:227:in `each'",
  "org/jruby/RubyArray.java:1613:in `each'", 
  "/home/logstash/logstash/vendor/bundle/jruby/1.9/gems/bindata-2.0.0/lib/bindata/array.rb:227:in `each'",
   "org/jruby/RubyEnumerable.java:758:in `collect'",
    "/home/logstash/logstash/lib/logstash/codecs/netflow/util.rb:50:in `get'",
    "/home/logstash/logstash/vendor/bundle/jruby/1.9/gems/bindata-2.0.0/lib/bindata/primitive.rb:116:in `sensible_default'",
     "/home/logstash/logstash/vendor/bundle/jruby/1.9/gems/bindata-2.0.0/lib/bindata/base_primitive.rb:151:in `_value'",
      "/home/logstash/logstash/vendor/bundle/jruby/1.9/gems/bindata-2.0.0/lib/bindata/primitive.rb:108:in `do_num_bytes'",
       "/home/logstash/logstash/vendor/bundle/jruby/1.9/gems/bindata-2.0.0/lib/bindata/struct.rb:317:in `sum_num_bytes_below_index'",
        "org/jruby/RubyRange.java:479:in `each'",
         "/home/logstash/logstash/vendor/bundle/jruby/1.9/gems/bindata-2.0.0/lib/bindata/struct.rb:314:in `sum_num_bytes_below_index'",
          "/home/logstash/logstash/vendor/bundle/jruby/1.9/gems/bindata-2.0.0/lib/bindata/struct.rb:309:in `sum_num_bytes_for_all_fields'",
           "/home/logstash/logstash/vendor/bundle/jruby/1.9/gems/bindata-2.0.0/lib/bindata/struct.rb:218:in `do_num_bytes'",
            "/home/logstash/logstash/vendor/bundle/jruby/1.9/gems/bindata-2.0.0/lib/bindata/base.rb:196:in `num_bytes'",
             "/home/logstash/logstash/lib/logstash/codecs/netflow.rb:182:in `decode'",
              "/home/logstash/logstash/vendor/bundle/jruby/1.9/gems/bindata-2.0.0/lib/bindata/array.rb:227:in `each'", 
              "org/jruby/RubyArray.java:1613:in `each'", "/home/logstash/logstash/vendor/bundle/jruby/1.9/gems/bindata-2.0.0/lib/bindata/array.rb:227:in `each'", 
              "/home/logstash/logstash/lib/logstash/codecs/netflow.rb:85:in `decode'", "/home/logstash/logstash/lib/logstash/inputs/udp.rb:100:in `inputworker'",
               "/home/logstash/logstash/lib/logstash/inputs/udp.rb:75:in `udp_listener'"], :level=>:error}

**Comment by jorritfolmer** [comment]
Line 50 of util.rb handles mac addresses, so it appears your netflow probes include mac addresses in their export. Could you provide a sample of this netflow traffic with something like tcpdump or wireshark, so I can try to replay it?

**Comment by alexchay** [comment]
This is a known bug. Here's how you can fix it. 
repeatedly/fluent-plugin-netflow@dfcd4f40e6bd57c1d08ddc6eae707fcdc52163f6

**Comment by jorritfolmer** [comment]
Nice! Thanks for the pointer!
I've been able to reproduce (and confirm the fix of) the original issue with:

```
nprobe -T "%IPV4_SRC_ADDR %IPV4_DST_ADDR %IN_SRC_MAC %OUT_DST_MAC" -n logstash.server.fqdn:2055
```

**Comment by jorritfolmer** [comment]
Closed with merge of #23
